### PR TITLE
Get tokens for shares compound yield

### PR DESCRIPTION
**Pull request by KalpitaMandal**
The exchangeRateCurrent is multiplied by the shares and a .div(1e18) is added because the exchangeRate is raised to 1e18 decimals in compound contracts

```
uint256 exchangeRateCurrent = ICToken(cToken).exchangeRateCurrent();
uint256 expectedamount = exchangeRateCurrent.mul(shares).div(1e18);
```

### CODE REVIEW

File: contracts/interfaces/Invest/ICToken.sol [hunk 1]
```diff
@@ -20,10 +20,10 @@ interface ICToken {
 
     function balanceOfUnderlying(address account) external returns (uint256);
 
-    function exchangeRateCurrent() external returns (uint256);
-
     function exchangeRateStored() external view returns (uint256);
```
Comment (prateekreddy) [inline contracts/interfaces/Invest/ICToken.sol:23]: Let's remove this if not being used.

File: contracts/yield/CompoundYield.sol [hunk 2]
```diff
@@ -224,7 +224,9 @@ contract CompoundYield is IYield, Initializable, OwnableUpgradeable, ReentrancyG
         //balanceOfUnderlying returns underlying balance for total shares
         if (shares == 0) return 0;
         address cToken = liquidityToken[asset];
-        uint256 amount = ICToken(cToken).balanceOfUnderlying(address(this)).mul(shares).div(IERC20(cToken).balanceOf(address(this)));
+        uint256 exchangeRateCurrent = ICToken(cToken).exchangeRateCurrent();
+        uint256 amount = exchangeRateCurrent.mul(shares).div(1e18);
+        // uint256 amount = ICToken(cToken).balanceOfUnderlying(address(this)).mul(shares).div(IERC20(cToken).balanceOf(address(this)));
```
Comment (prateekreddy) [inline contracts/yield/CompoundYield.sol:229]: Cleanup comments

**Review by prateekreddy — CHANGES_REQUESTED** [review]

File: contracts/yield/CompoundYield.sol [hunk 3]
```diff
@@ -237,10 +239,11 @@ contract CompoundYield is IYield, Initializable, OwnableUpgradeable, ReentrancyG
      **/
     function getTokensForShares(uint256 _shares, address _asset) public override returns (uint256) {
         //balanceOfUnderlying returns underlying balance for total shares
-        if (_shares == 0) return 0;
-        address _cToken = liquidityToken[_asset];
-        uint256 _amount = ICToken(_cToken).balanceOfUnderlying(address(this)).mul(_shares).div(IERC20(_cToken).balanceOf(address(this)));
-        return _amount;
+        if (shares == 0) return 0;
+        address cToken = liquidityToken[asset];
+        uint256 exchangeRateCurrent = ICToken(cToken).exchangeRateCurrent();
```
Comment (prateekreddy) [inline contracts/yield/CompoundYield.sol:244]: Use exchangeRateCurrent for getSharesForTokens as well

**Review by prateekreddy — CHANGES_REQUESTED** [review]

File: contracts/yield/CompoundYield.sol (hunk 3 above)
Comment (KalpitaMandal) [reply contracts/yield/CompoundYield.sol:242]: Added the changes

File: contracts/yield/CompoundYield.sol [hunk 4]
```diff
@@ -12,6 +12,8 @@ import '../interfaces/Invest/ICToken.sol';
 import '../interfaces/Invest/Comptroller.sol';
 import '../interfaces/IWETH9.sol';
 
+import 'hardhat/console.sol';
```
[the remark this answers is not in the document]
Comment (KalpitaMandal) [reply contracts/yield/CompoundYield.sol]: Removed

File: contracts/yield/CompoundYield.sol [hunk 5]
```diff
@@ -250,7 +251,9 @@ contract CompoundYield is IYield, Initializable, OwnableUpgradeable, ReentrancyG
      * @return amount of shares for given tokens
      **/
     function getSharesForTokens(uint256 _amount, address _asset) public override returns (uint256) {
-        return (_amount.mul(1e18)).div(getTokensForShares(1e18, _asset));
+        address cToken = liquidityToken[_asset];
+        uint256 exchangeRateCurrent = ICToken(cToken).exchangeRateCurrent();
+        return (_amount.mul(1e18)).div(exchangeRateCurrent);
```
Comment (prateekreddy) [inline contracts/yield/CompoundYield.sol:256]: .mul(1e18) no necessary

**Pull request merged**